src/transport.js

```diff
@@ -53,11 +54,18 @@ module.exports = function (swarm) {
       // specific for the transport we are using
       const proxyConn = new Connection()
 
-      next(multiaddrs.shift())
+      each(multiaddrs, next)
```

Comment:
@jackkleeman @dignifiedquire I'm deep into libp2p for the circuit-relay stuff right not, so I can squeeze this in, how quickly do you need this @jackkleeman? Is this something thats blocking you?